pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -0,0 +1,374 @@
+	if !ret {
+		klog.InfoS("Pod does not have a NetworkAttachmentDefinition", "Pod", klog.KObj(pod))
+		return nil
+	}
+	// get Pod/Container information (container id, namespace, etc.,) from CNI info cache (added by CNI Server CmdAdd upon
+	// successful Pod and primary network creation).
+	// If container information is not available yet in the CNICache, return error to requeue work
+	podCNIInfo := pc.podCache.GetCNIConfigInfoPerPod(pod.Name, pod.Namespace)
+	if len(podCNIInfo) == 0 {
+		return errors.New("CNI Cache Information not found")
+	} else {
+		networklist, err := parsePodSecondaryNetworkAnnotation(secondaryNetwork)
+		if err != nil {
+			return errors.New("secondary network annotation parsing failed")
+		}
+		pc.configureSecondaryNetwork(pod, networklist, podCNIInfo)
+	}
+	return nil
+}
+
+func (pc *PodController) handleRemovePod(obj interface{}) error {
+	pod := obj.(*corev1.Pod)
+	// Read the CNI info (stored during Pod creation by cniserver) from cache.
+	// Delete CNI info shared in cache for a specific Pod which is getting removed/deleted.
+	podCNIInfo := pc.podCache.GetCNIConfigInfoPerPod(pod.Name, pod.Namespace)
+	if len(podCNIInfo) > 0 {
+		for _, containerInfo := range podCNIInfo {
+			// delete Pod information from CNI info cache.
+			pc.podCache.DeleteCNIConfigInfo(containerInfo)
+		}
+	}
+	return nil
+}
+
+func (pc *PodController) syncPod(key string) error {
+	obj, exists, err := pc.podInformer.GetIndexer().GetByKey(key)
+	if err != nil {
+		return err
+	} else if exists {
+		return pc.handleAddUpdatePod(obj)
+	} else {
+		return pc.handleRemovePod(obj)
+	}
+}
+
+func (pc *PodController) Worker() {
+	for pc.processNextWorkItem() {
+	}
+}
+
+func (pc *PodController) processNextWorkItem() bool {
+
+	obj, quit := pc.queue.Get()
+	if quit {
+		return false
+	}
+	defer pc.queue.Done(obj)
+	if key, ok := obj.(string); !ok {
+		pc.queue.Forget(obj)
+		return true
+	} else if err := pc.syncPod(key); err == nil {
+		klog.InfoS("Successfully processed key", key, "in queue")
+		pc.queue.Forget(key)
+	} else {
+		klog.InfoS("Failed syncing Pod. retry shortly", key, "Error", err)
+	}
+	return true
+}
+
+// Configure SRIOV VF as a Secondary Network Interface.
+func (pc *PodController) configureSriovAsSecondaryInterface(pod *corev1.Pod, netinfo *SecondaryNetworkObject, containerInfo *cnipodcache.CNIConfigInfo, result *current.Result) error {
+	podSriovVFDeviceID, err := cniserver.GetPodContainerDeviceIDs(pod.Name, pod.Namespace)
+	if err != nil {
+		return fmt.Errorf("GetPodContainerDeviceIDs failed: %v", err)
+	}
+
+	if err = pc.cniServer.GetPodConfigurator().ConfigureSriovSecondaryInterface(
+		containerInfo.PodName,
+		containerInfo.PodNameSpace,
+		containerInfo.ContainerID,
+		containerInfo.ContainerNetNS,
+		netinfo.InterfaceName,
+		containerInfo.MTU,
+		podSriovVFDeviceID[0],
+		result,
+	); err != nil {
+		return fmt.Errorf("SRIOV Interface creation failed. %v", err)
```

Comment:
Nit: sometime your use ":" to append the previous error, sometimes ".". I generally suggest ":".